pulp_rpm/app/tasks/copy.py

```diff
@@ -90,20 +90,31 @@ def find_children_of_content(content, src_repo_version):
             packagegroups = packagegroups.union(opt_env_package_groups)
 
     # Find rpms referenced by PackageGroups
+    group_package_names = set()
     for packagegroup in packagegroups.iterator():
```

Comment:
Maybe "simpler" is the wrong word but it should use way less queries.

Reply:
With PackageEnvironments and PackageCategories the difference is a lot smaller and less relevant (because there's less of them)